acmwebsite/controllers/wiki.py

```diff
@@ -0,0 +1,99 @@
+"""Wiki controller module"""
+from tg import expose
+
+from acmwebsite.lib.base import BaseController
+
+from pygit2 import Repository,init_repository
+from pygit2 import Tree
+from pygit2 import Signature
+import pygit2 as pg
+
+from docutils.core import publish_parts
+
+import os
+import tg
+
+__all__ = ['WikiController']
+
+class WikiController(BaseController):
+    """Controls the wiki"""
+
+    def _before(self, *args, **kw):
+        try:
+            repo_path = tg.config.get('wiki.repo')
+            if not repo_path:
+                tg.abort(400, "Wiki not enabled")
+            self.repo = Repository(repo_path)
+        except pg.GitError:
+            self._init_wiki_repo()
+
+    def _init_wiki_repo(self):
+        self.repo = init_repository(tg.config.get('wiki.repo'), True)
+        signature = Signature("Example McPersonson", "emailuser@emailsite.tld") #TODO: Replace me
```

Comment:
Explain?

Reply:
Explain what - the TODO?

If so - I just mean that this Signature can be replaced if need be, though it is only used for the inital commit. Presumably page commits using the web editor (once implemented) will pull signature details from the logged in profile.